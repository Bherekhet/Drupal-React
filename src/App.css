.home {
  background: #ecf0f1;
}

.mainContent {
  width: 80%;
  margin: 0 auto;
  text-align: center;
  margin-top: 10px;
  margin-bottom: 50px;
}

.card {
  width: 60%;
  margin: 0 auto;
  margin-bottom: 10px;
}

.recipeDetail {
  margin: 0 auto;
  margin: 5%;
}

.detail{
  margin-top: 2.2em;
}

.recipeTitle {
  font-size: 1.5em;
  text-align: center;
}

.imgNav{
  display: block;
}

.next {
  float: right;
}

