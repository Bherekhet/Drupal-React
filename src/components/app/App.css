.home {
    background: #FBFCFC;
}

.footer {
    background: #543864;
    color: white;
    clear: both;
    width: 100%;
}

.footer a {
    color: grey;
}