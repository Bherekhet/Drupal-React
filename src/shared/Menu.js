import '../../node_modules/bootstrap/dist/css/bootstrap.min.css';
import React from 'react';
import '../components/home/Home.css';
import {Link} from "react-router-dom";

export default function Menu (){
    return (
      <div>
          <nav id="navColor" className="navbar navbar-expand-md navbar-dark ">
            <a className="navbar-brand" href="#" >Recipe Magazine</a>
             <ul className="navbar-nav">
               <li className="nav-item">
                 <a className="nav-link"  href="#">Home</a>
               </li>
                <li className="nav-item">
                 <a className="nav-link" href="#">Recipes</a>
                </li>
             </ul>
          </nav>
      </div>
    )
  }