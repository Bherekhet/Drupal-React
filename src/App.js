import React from 'react';
import './App.css';
import lorem from './shared/Lorem-ipsum';
import Footer from './shared/Footer';
import Menu from './shared/Menu';

// function Control(){
//   fetch("http://gtest.dev.wwbtc.com/json/rec/").then(response=> 
//      // console.log('res'+response.json());
//       response.json())
//     .then(data => {
//       return data;
//     }).catch(err => {
//       this.console.log(err);
//     })
// }

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {recipeList: [], }
    
    this.fetchRecipe = this.fetchRecipe.bind(this);
  }

  fetchRecipe(){
    fetch("http://gtest.dev.wwbtc.com/json/rec/").then(response=> {
      return response.json();
    }).then(data => {
      this.setState({recipeList: data});
    }).catch(err => {
      this.console.log(err);
    })
  }

  componentWillMount(){
    this.fetchRecipe();
  }

  render(){
    const {recipeList} = this.state;
    //console.log(recipeList);
    if (!recipeList){
      this.fetchRecipe();
    }

    return(
      <div className="container">
        <Menu />
        <Home list={recipeList} />
        <Footer />
      </div>
    );
  }
}

class Home extends React.Component {
  constructor(props){
    super(props);

    this.state = {
      clicked: false
    };

    this.onRecipeClick = this.onRecipeClick.bind(this);
  }

  onRecipeClick(id){
    this.setState({
      clicked: true,
      id: id,
    });
  }
  
  render(){
    const {clicked} = this.state;
    const list = this.props.list;
    if (!list){
      return null;
    }

    if (clicked){
      return <Recipe recipeID={this.state.id} recipeList={list}/>
    } 
    return( 
      <div className="home">
        <div className="mainContent">
          {lorem.generateParagraphs(3)}
        </div>
          { list.map((item, index) => 
          <div className="grid">
             <div className="card" key={index}>
              {console.log('http://gtest.dev.wwbtc.com'+item.field_images)}

              <a href="#" onClick={() => this.onRecipeClick(item.nodeID)}>
              <img className="card-img-top imgSize" src={'http://gtest.dev.wwbtc.com'+item.field_images} alt={item.title}/>
              </a>
              <div className="card-body">
                <h5 className="card-title">{item.title}</h5>
                <p className="card-text">{item.field_summary}</p>
              </div>
            </div>
          </div>
          )}
      </div>
    );
  }
}

class Recipe extends React.Component {
  render(){
    return(
      <div>
          <p className="recipeTitle">Doro Wat</p>
          <div className="card">
            <img className="rounded mx-auto d-block" style={{width: '80%'}} 
              src="https://amboethiopiancuisine.files.wordpress.com/2011/06/013011-035.jpg?w=1800&h=720&crop=1)">
              </img>
            <div className="card-body">
              <button style={{width: "100px"}} className="btn btn-primary previous">Previous</button>
              <button style={{width: "100px"}} className="btn btn-primary next">Next</button>
            </div>
          </div>
          <p className="detail">{lorem.generateSentences(8)}</p>
          <p className="recipeTitle">What you will need to make the dish</p>
          <ul>
            <li>{lorem.generateSentences(1)}</li>
            <li>{lorem.generateSentences(1)}</li>
            <li>{lorem.generateSentences(1)}</li>
            <li>{lorem.generateSentences(1)}</li>
            <li>{lorem.generateSentences(1)}</li>
          </ul>
      </div>
    )
  }
}

export default App;
