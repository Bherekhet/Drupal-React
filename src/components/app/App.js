import React from 'react';
import Footer from '../../shared/Footer';
import Menu from '../../shared/Menu';
import Home from '../home/Home';

//this component will set the general look of the page
class App extends React.Component {
  constructor() {
    super();
    this.state = {recipeData: [], homePage: [] }
    this.fetchRecipe = this.fetchRecipe.bind(this);
    this.fetchHomePage = this.fetchHomePage.bind(this);
  }

  //get all recipes from given API endpoint
  fetchRecipe(){
    fetch(process.env.REACT_APP_API_BASE+process.env.REACT_APP_API_RECIPE_ENDPOINT).then(response=> {
      return response.json();
    }).then(data => {
      this.setState({recipeData: data});
    }).catch(err => {
      this.console.log(err);
    })
  }
  //gets the home page info from give API endpoint
  fetchHomePage(){
    fetch(process.env.REACT_APP_API_BASE+process.env.REACT_APP_API_HOME_ENDPOINT).then(response => {
      return response.json();
    }).then(data => {
      this.setState({homePage: data});
    }).catch(err => {
      this.console.log(err);
    })
  }

  componentDidMount(){
    this.fetchRecipe();
    this.fetchHomePage();
  }

  render(){
    const {recipeData, homePage} = this.state;

    return(
      <div>
        <Menu/>
        <Home recipeData={recipeData} homePage={homePage}/>
        <Footer/>

        {/* tried to implement react router, did not work completely */}
        {/* <Router>
          <Menu/>
          <Switch>
            <Route path="/" exact render={(props) => <Home {... props} recipeClicked="false" recipeData={recipeData} homePage={homePage}/>}/>/>
            <Route path="/recipes" exact render={(props) => <Recipe {... props} recipeID={0} recipeData={recipeData}/>}/>/>
            <Route path="/recipes/:id" render={(props) => <Home {... props}  recipeClicked="true" recipeData={recipeData} homePage={homePage}/>}/>/>
          </Switch>
          <Footer />
        </Router> */}
      </div> 
    );
  }

}

export default App;
