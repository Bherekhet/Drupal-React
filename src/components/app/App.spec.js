import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import App from './App.js';
import Menu from '../../shared/Menu.js'
import ReactDOM from 'react-dom';
import { JestEnvironment } from '@jest/environment';

jest.mock("react-dom", () => ({render: jest.fn()}))

describe('App component', () => {
  
  test('snapshot renders App.js', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Nav component', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find(Menu).length).toEqual(1);
  });

  //snapshot test index.js
  it('renders index.js without crashing', () => {
    const div = document.createElement('div');
    div.id = "root";
    document.body.appendChild(div);
    require("../../../src/index.js");
    expect (ReactDOM.render).toHaveBeenCalledWith(<App/>, div);
  });
})