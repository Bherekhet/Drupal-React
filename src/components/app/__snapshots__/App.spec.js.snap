// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot renders App.js 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-md navbar-dark "
      id="navColor"
    >
      <a
        className="navbar-brand"
        href="#"
      >
        Recipe Magazine
      </a>
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="#"
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="#"
          >
            Recipes
          </a>
        </li>
      </ul>
    </nav>
  </div>
  <div
    className="main"
  >
    <p
      className="intro"
    >
      Oops no data found for homepage on content manager
    </p>
    <div
      className="contain"
    >
      <p>
        Oops no data found for recipes on content manager
      </p>
    </div>
  </div>
  <div
    className="footer"
  >
    <div
      className="footer-copyright text-center py-3 "
    >
      Â© 2019 Copyright:
      <a
        href="https://github.com/Bherekhet/Drupal-React.git"
      >
         Bereket Degefa
      </a>
    </div>
  </div>
</div>
`;
